staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/subresource.go

```diff
@@ -113,17 +113,24 @@ func (h *ScaleHandler) ToParent(scaleEntries []metav1.ManagedFieldsEntry) ([]met
 			continue
 		}
 
-		// "Steal" the replicas path from the main resource entry
-		newSet := versionedSet.Set().Difference(fieldpath.NewSet(path))
-
-		if !newSet.Empty() {
-			newVersionedSet := fieldpath.NewVersionedSet(
-				newSet,
-				versionedSet.APIVersion(),
-				versionedSet.Applied(),
-			)
-			f[manager] = newVersionedSet
+		if _, ok := scaleFields[manager]; !ok {
+			// "Steal" the replicas path from the main resource entry
+			newSet := versionedSet.Set().Difference(fieldpath.NewSet(path))
+
+			if !newSet.Empty() {
+				newVersionedSet := fieldpath.NewVersionedSet(
+					newSet,
+					versionedSet.APIVersion(),
+					versionedSet.Applied(),
+				)
+				f[manager] = newVersionedSet
+				t[manager] = decodedParentEntries.Times()[manager]
+			}
+		} else {
+			// Field wasn't stolen, let's keep the entry as it is.
+			f[manager] = versionedSet
 			t[manager] = decodedParentEntries.Times()[manager]
+			delete(scaleFields, manager)
```

Comment:
I am not sure I fully understand this change. The keys of  scaleFields will always have the `-scale` suffix so the parent manager will match only i it has the same manager name and it has Subresource field set to scale. In this case, it's OK to steal I think. The result is the same but the code is more complicated.

Reply:
No, because there could be a remaining entry from when we extracted entries from the parent resource in the first place.

Reply:
right, agree, it's an edge case that I missed.